src/irmin-test/store.ml

```diff
@@ -194,7 +194,11 @@ module Make (S : S) = struct
       check_val "v1" (Some v1) v1';
       P.Contents.find t kv2 >>= fun v2' ->
       check_val "v2" (Some v2) v2';
-      P.Repo.close repo
+      P.Repo.close repo >>= fun () ->
+      try
+        ignore (with_contents repo (fun t -> P.Contents.add t v2));
```

Comment:
I am never sure what is the interaction between `ignore` and `lwt` ; in that case I believe it means that `P.Contents.add` will run in the background -- if it becomes stuck on IO for any reason, `Alcotest.fail` will then execute.

To make sure this won't happen you should probably use `>>=` to force the sequentiality of events (and `Lwt.catch` instead of `try ... with`)